src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -516,7 +528,7 @@ def process_vmss_create_namespace(namespace):
     validate_location(namespace)
     _validate_vm_create_storage_profile(namespace, for_scale_set=True)
     _validate_vmss_create_load_balancer(namespace)
-    _validate_vm_create_vnet(namespace)
+    _validate_vm_create_vnet(namespace, True)
```

Comment:
For clarity, I would make the kwarg `for_scale_set` explicit here like you did above.